vue/components/ui/molecules/bezier-curve/bezier-curve.vue

```diff
@@ -0,0 +1,377 @@
+        /**
+         * Bezier curve endpoint 2 Y coordinate.
+         */
+        y2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 X coordinate.
+         */
+        cx1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 1 Y coordinate.
+         */
+        cy1: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 X coordinate.
+         */
+        cx2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * Bezier curve control point 2 Y coordinate.
+         */
+        cy2: {
+            type: Number,
+            required: true
+        },
+        /**
+         * The number of pixels of the svg viewport
+         * in the format "`width`x`height`".
+         */
+        resolution: {
+            type: String,
+            default: "1024x1024"
+        },
+        /**
+         * Visibilty of Bezier curve control points.
+         */
+        showPoints: {
+            type: Boolean,
+            default: true
+        },
+        /**
+         * The interval at which the new coordinates
+         * will be emited when dragging.
+         */
+        emitInterval: {
+            type: Number,
+            default: 250
+        },
+        /**
+         * The decimal places to define the numerical precision
+         * when emitting the position of control and end points.
+         */
+        roundPlaces: {
+            type: Number,
+            default: 2
+        },
+        /**
+         * Weather or not the control points are disabled
+         * and unable to control.
+         */
+        disabled: {
+            type: Boolean,
+            default: false
+        }
+    },
+    data: function() {
+        return {
+            bezier: {
+                x1: this.x1,
+                y1: this.y1,
+                x2: this.x2,
+                y2: this.y2,
+                cx1: this.cx1,
+                cx2: this.cx2,
+                cy1: this.cy1,
+                cy2: this.cy2
+            },
+            dragging: null,
+            reactiveEmit: null
+        };
+    },
+    watch: {
+        x1(value) {
+            this.$set(this.bezier, "x1", value);
+        },
+        y1(value) {
+            this.$set(this.bezier, "y1", value);
+        },
+        x2(value) {
+            this.$set(this.bezier, "x2", value);
+        },
+        y2(value) {
+            this.$set(this.bezier, "y2", value);
+        },
+        cx1(value) {
+            this.$set(this.bezier, "cx1", value);
+        },
+        cy1(value) {
+            this.$set(this.bezier, "cy1", value);
+        },
+        cx2(value) {
+            this.$set(this.bezier, "cx2", value);
+        },
+        cy2(value) {
+            this.$set(this.bezier, "cy2", value);
+        },
+        dragging(value) {

```

Comment:
I believe this must be changed :/

Avoid using drag as emit reference, because conceptually there might be other things changing the coordinates (stuff like keyboard shortcuts or whatever). This is why we usually place the emitter of variable X in the watcher of variable X. You can still use a timeout with this to get the throttling you want

This way, you change the logic from "emit updates while dragging" to "emit updates to when the variables change". Right now they seem like the same thing, but semantically I don't think it is 100% correct